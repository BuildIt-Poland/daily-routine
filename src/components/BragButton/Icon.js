import React from 'react';
import styled from 'styled-components';

import { Lightbulb } from '../Icons';

const Wrapper = styled.div`
  position: absolute;
  right: 0;
  bottom: -1rem;
`;

function Icon() {
  return (
    <Wrapper>
      <Lightbulb />
    </Wrapper>
  );
}

export default Icon;
