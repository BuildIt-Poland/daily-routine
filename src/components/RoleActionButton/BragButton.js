import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { colorGreen, colorDarkGreen } from '../../styles/designTokens';
import Button from '../Button';
import LightbulbIcon from './LightbulbIcon';

const StyledButton = styled(Button)`
  background-color: ${colorGreen};
  border-bottom-color: ${colorDarkGreen};
`;

function BragButton({ to }) {
  return (
    <StyledButton as={Link} to={to}>
      Brag about my efforts
      <LightbulbIcon />
    </StyledButton>
  );
}

BragButton.propTypes = {
  to: PropTypes.string.isRequired
};

export default BragButton;
