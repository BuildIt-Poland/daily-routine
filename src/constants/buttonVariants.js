export const BRAG = 'brag';
export const CONFESS = 'confess';
export const FRONT_END = 'frontend';
export const BACK_END = 'backend';
export const DEV_OPS = 'devops';
