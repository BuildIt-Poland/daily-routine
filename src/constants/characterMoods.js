export const CONFUSED = 'confused';
export const CONFIDENT = 'confident';
export const SAD = 'sad';
