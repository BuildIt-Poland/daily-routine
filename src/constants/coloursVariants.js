export const FRONT_END_VARIANT = 'frontend';
export const BACK_END_VARIANT = 'backend';
export const DEV_OPS_VARIANT = 'devops';
export const CONFESS_VARIANT = 'confess';
export const BRAG_VARIANT = 'brag';
