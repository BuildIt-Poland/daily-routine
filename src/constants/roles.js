// Roles
export const FRONT_END_ROLE = 'frontend';
export const BACK_END_ROLE = 'backend';
export const DEV_OPS_ROLE = 'devops';

// Role actions
export const BRAG_ROLE_ACTION = 'brag';
export const CONFESS_ROLE_ACTION = 'confess';
