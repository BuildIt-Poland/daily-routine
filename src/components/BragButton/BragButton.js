import React from 'react';

import { BRAG } from '../../constants/buttonVariants';
import Button from '../Button';
import Icon from './Icon';

function BragButton() {
  return (
    <Button variant={BRAG} to="/" icon={<Icon />}>
      Brag about my efforts
    </Button>
  );
}

export default BragButton;
