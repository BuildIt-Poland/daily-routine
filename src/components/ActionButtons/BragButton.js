import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { role } from '../../types';
import { BRAG_ROLE_ACTION } from '../../constants/roles';
import { colorGreen, colorDarkGreen } from '../../styles/designTokens';
import LightbulbIcon from './LightbulbIcon';
import ButtonBase from './ButtonBase';

const StyledButton = styled(ButtonBase)`
  background-color: ${colorGreen};
  border-bottom-color: ${colorDarkGreen};
`;

const BUTTON_LABEL = 'Brag about my efforts';
const UPDATED_BUTTON_LABEL = 'Brag more';

function BragButton({ quoteID, role, ...props }) {
  const [text, setText] = useState(BUTTON_LABEL);

  const updateButtonLabel = () => setText(UPDATED_BUTTON_LABEL);

  return (
    <StyledButton
      {...props}
      to={`/${role}/${BRAG_ROLE_ACTION}/${quoteID}`}
      icon={LightbulbIcon}
      onClick={updateButtonLabel}
    >
      {text}
    </StyledButton>
  );
}

BragButton.propTypes = {
  role: role.isRequired,
  quoteID: PropTypes.string.isRequired
};

export default BragButton;
