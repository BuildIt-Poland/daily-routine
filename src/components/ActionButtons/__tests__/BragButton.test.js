import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent, cleanup } from '@testing-library/react';

import { FRONT_END_ROLE } from '../../../constants/roles';
import BragButton from '../BragButton';

afterEach(cleanup);

describe('COMPONENT - RoleButton BragButton', () => {
  it('should contain text `Brag about my efforts`', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <BragButton role={FRONT_END_ROLE} />
        tl goo
      </MemoryRouter>
    );

    expect(container.querySelector('a')).toHaveTextContent('Brag about my efforts');
  });

  it('should contain `svg` element', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <BragButton role={FRONT_END_ROLE} />
      </MemoryRouter>
    );

    expect(container.contains(container.querySelector('svg'))).toBe(true);
  });

  it('after once click should change text to `Brag more`', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <BragButton role={FRONT_END_ROLE} />
      </MemoryRouter>
    );

    const bragButton = container.querySelector('a');

    fireEvent.click(bragButton);

    expect(bragButton).toHaveTextContent('Brag more');
  });
});
