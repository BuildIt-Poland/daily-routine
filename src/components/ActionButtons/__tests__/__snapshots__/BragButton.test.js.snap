// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - RoleButton BragButton renders correctly 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #21c230;
  color: #ffffff;
  padding: 0 2rem;
  margin: 2rem 0;
  border: 0;
  border-bottom: 0.3rem solid #138a1e;
  border-radius: 0.6rem;
  font-size: 1.8rem;
  font-weight: 700;
  text-align: left;
  width: 100%;
<<<<<<< HEAD
  height: 7rem;
  background-color: #21c230;
  border-bottom-color: #138a1e;
=======
  height: 5rem;
>>>>>>> master
}

.c1:not(:disabled) {
  cursor: pointer;
}

<<<<<<< HEAD
@media (max-width:26.25em) {
  .c0 {
    height: 5rem;
  }
=======
.c0 {
  background-color: #21c230;
  border-bottom-color: #138a1e;
>>>>>>> master
}

<Link
  className="c0 c1"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "borderBottomWidth": 3,
    }
  }
  to="/frontend/brag"
>
  Brag about my efforts
  LightbulbIcon
</Link>
`;
